src/Traits/SocialLoginTrait.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Canvas\Traits;
+
+use function Canvas\Core\envValue;
+use function time;
+use Canvas\Models\Sources;
+use Canvas\Models\Users;
+use Canvas\Models\UserLinkedSources;
+use Phalcon\Security\Random;
+
+
+/**
+ * Trait SocialLoginTrait.
+ *
+ * @package Niden\Traits
+ */
+trait SocialLoginTrait
+{
+    /**
+     * Facebook Login
+     * @param Sources $source
+     * @param string $accessToken
+     * @return Array
+     */
+    protected function facebookLogin(Sources $source, string $accessToken): Array
+    {
+
+        /**
+         * Get the Facebook adapter
+         */
+        $facebookAdapter = $this->di->get('socialLogin')->authenticate(ucfirst($source->title));
+
+        /**
+         * Set user's Access Token
+         */
+        $facebookAdapter->setAccessToken($accessToken);
+
+        /**
+         * Get user's profile based on set Access Token
+         */
+        $data = $facebookAdapter->getUserProfile();
+
+
+        /**
+         * Lets Login or Signup the user
+         */
+        $userProfile = current($data);
+
+        /**
+        * Lets find if user has a linked source by social network id
+        */
+        $userLinkedSource =  UserLinkedSources::findFirst([
+            'conditions'=>'source_id = ?0 and source_users_id_text = ?1 and is_deleted = 0',
+            'bind'=>[$source->id,$userProfile->identifier]
+        ]);
+
+        /**
+         * Verify if user exists
+         */
+        $user = Users::findFirst([
+            'conditions' => 'email = ?0 and is_deleted = 0',
+            'bind' => [$userProfile->email]
+        ]);
+
+        if (!is_object($user) && !is_object($userLinkedSource)) // User does not exist and has not been linked to a source
```

Comment:
Do not make if statements that long. It is harder to read for a dev.
What about flipping the statement?

if ($user && $userLinkedSource){
	return $this->loginSocial($user);
}

// Here you don't need the else anymore.

// Registering the user